### ISSUE TITLE
[FEATURE] Add a category/role navigation bar in staff assignment for staff filtering 

### ISSUE THREAD

**Issue by RaymondTMP** [COMP3122-2324-Project/Simpsyber#21]
Currently, players need to click the arrow button to assign staff, especially tester, for a task. Since it is a repeating action for players, players tend to perform this action faster. It is prone to human error. Since staff reassignment is not available and more staff are expected to be available in later stages of the game, adding a navigation bar for filtering is a solution to improve player experiences. 
![Screenshot 2024-02-29 113424](https://github.com/COMP3122-2324-Project/Simpsyber/assets/74492911/c64318a3-e36a-4a94-ae69-37b0cc51d150)

**Comment by ln2-muyuan** [comment]
I agree with you, and I'm also proposing to set up a filter that directly selects the fastest employee. Also, in the current game settings, if an employee is already occupied, you can't see what time he needs to spend on the next job. I think this restriction should be removed so that the player can better allocate his staff without having to wait for the staff to be free to check the time he needs.

**Comment by chungfg** [comment]
I think that is a great idea. Currently, because of the size of the card, one page can only include four staffs. Other than a naviagation bar proposed by @RaymondTMP, I think another straight forward alternative on this is shrinking the size of the employee card in order to include them in one page. Then, the player could select all six staff in one click.

**Comment by ChowSingYeung** [comment]
I think all of your suggestions are valid and helpful with the user experience, I think other way to help improve this is to add sorting to the staff assignment menu, user may choose to sort by avaliability, or by the shortest time to finish the task to make assigning staff easier for the user.
